src/Interactivity/woo-directives.php

```diff
@@ -0,0 +1,123 @@
+<?php
+include_once __DIR__ . '/../../../gutenberg/lib/experimental/html/wp-html.php';
+
+require_once __DIR__ . '/directives/class-woo-directive-context.php';
+require_once __DIR__ . '/directives/class-woo-directive-store.php';
+require_once __DIR__ . '/directives/woo-process-directives.php';
+
+require_once __DIR__ . '/directives/attributes/woo-bind.php';
+require_once __DIR__ . '/directives/attributes/woo-class.php';
+require_once __DIR__ . '/directives/attributes/woo-style.php';
+require_once __DIR__ . '/directives/tags/woo-context.php';
+
+
+/**
+ * Register the Interactivity API scripts. These files are enqueued when a block
+ * defines `woo-directives-runtime` as a dependency.
+ */
+function woo_directives_register_scripts() {
+	wp_register_script(
+		'woo-directives-vendors',
+		plugins_url( '../../build/woo-directives-vendors.js', __FILE__ ),
+		array(),
+		'1.0.0',
+		true
+	);
+	wp_register_script(
+		'woo-directives-runtime',
+		plugins_url( '../../build/woo-directives-runtime.js', __FILE__ ),
+		array( 'woo-directives-vendors' ),
+		'1.0.0',
+		true
+	);
+}
+add_action( 'init', 'woo_directives_register_scripts' );
+
+
+function woo_directives_get_client_side_navigation() {
+	static $client_side_navigation = null;
+	if ( is_null( $client_side_navigation ) ) {
+		$client_side_navigation = apply_filters( 'client_side_navigation', false );
+	}
+	return $client_side_navigation;
+}
+
+function woo_directives_add_client_side_navigation_meta_tag() {
+	if ( woo_directives_get_client_side_navigation() ) {
+		echo '<meta itemprop="woo-client-side-navigation" content="active">';
+	}
+}
+add_action( 'wp_head', 'woo_directives_add_client_side_navigation_meta_tag' );
+
+
+function woo_directives_mark_interactive_blocks( $block_content, $block, $instance ) {
+	if ( woo_directives_get_client_side_navigation() ) {
+		return $block_content;
+	}
+
+	// Append the `data-woo-ignore` attribute for inner blocks of interactive blocks.
+	if ( isset( $instance->parsed_block['isolated'] ) ) {
+		$w = new WP_HTML_Tag_Processor( $block_content );
+		$w->next_tag();
+		$w->set_attribute( 'data-woo-ignore', '' );
```

Comment:
I think the preferred way of doing this is

```suggestion
		$w->set_attribute( 'data-woo-ignore', true );
```

The HTML Tag Processor is smart enough to translate that to

```html
<some-tag data-woo-ignore>
```

(and not `<some-tag data-woo-ignore="true">`, if that was your concern 😄 ).

Reply:
Ah, nice! In that case, I think we'd have to change that in our repo as well (I just copied the code from there).